shared-module/board/__init__.c

```diff
@@ -148,12 +153,19 @@ void reset_board_busses(void) {
     bool display_using_spi = false;
     #if CIRCUITPY_DISPLAYIO
     for (uint8_t i = 0; i < CIRCUITPY_DISPLAY_LIMIT; i++) {
-        if (displays[i].fourwire_bus.bus == spi_singleton) {
+        mp_const_obj_t bus_type = displays[i].bus_base.type;
+        if (bus_type == &displayio_fourwire_type && displays[i].fourwire_bus.bus == spi_singleton) {
```

Comment:
This is ready for a fresh look!  Instead of adding a special case I removed one, so it's sure to be right(-er) this time.

@tannewt Is this specific change (to check the type of the bus before checking its bus property) right?  I assume it's right (because if it's, say an I2CDisplay then the content at that address is unrelated but also would never equal the spi_singleton) but if it's intentional I'd change it back and add a comment explaining it.

Reply:
Ya, this looks correct to me. We check the type to know which union entry to use.